### PR TITLE
v1.12 backport - Enable Google Analytics 4

### PR DESCRIPTION
**Pull request by qmonnet**
Backport of https://github.com/cilium/cilium/pull/22220, to enable Google Analytics on the v1.12 branch.

```release-note
docs: Enable Google Analytics for v1.12 documentation
```

Cc @chalin, @joestringer

We'll need a new docs-builder image.

---

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 21069 21040 22220; do contrib/backporting/set-labels.py $pr done 1.12; done
```

### CODE REVIEW

**Review by nebril — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by qmonnet** [comment]
Documentation workflow is now failing
```
Running linter...
Validating documentation (syntax, spelling)...

Please fix the following documentation warnings:
WARNING: Called: git log --quiet --no-color --pretty=format:%(trailers:key=Co-Authored-By,separator=%x0A)%x0A%an%x0A%cn
WARNING: Failed to scan contributors: Command '['git', 'log', '--quiet', '--no-color', '--pretty=format:%(trailers:key=Co-Authored-By,separator=%x0A)%x0A%an%x0A%cn']' returned non-zero exit status 128.
```

I can reproduce locally with the new image, but can't reproduce when I generate a docs-builder image myself :thinking:

**Comment by joestringer** [comment]
Is it possible there's some local filesystem state that you inadvertently introduce into the builder image when building locally?

**Comment by qmonnet** [comment]
Apparently yes; testing on a fresh fork, I can reproduce when building the image locally, too, from the tip of v1.12.

And I found the cause:

```
docker run --workdir /github/workspace --rm -v "$PWD/..":"/github/workspace" -it cilium/docs-builder:latest /bin/bash                                         
bash-5.1# git log --quiet --no-color --pretty='format:%(trailers:key=Co-Authored-By,separator=%x0A)%x0A%an%x0A%cn'
fatal: detected dubious ownership in repository at '/github/workspace'
To add an exception for this directory, call:

        git config --global --add safe.directory /github/workspace
```

I'll backport the commit from https://github.com/cilium/cilium/pull/21069.

**Comment by qmonnet** [comment]
@joestringer This Should Be Good :tm:. Would you mind generating a new image from this PR, pretty-please?

**Comment by joestringer** [comment]
:heavy_check_mark: triggered here, same tag name for today's date + v1.12: https://hub.docker.com/repository/docker/cilium/docs-builder/builds

**Comment by qmonnet** [comment]
I had to backport https://github.com/cilium/cilium/commit/ba640b34ffa843e00dc3b75cacedcf5ea1ed957a as well, which thankfully does not require yet another image (phew!) as it's only touching check-build.sh, not included in the image.

Doc workflow is now passing, I'll wait for Travis to complete and we should be good to go.

**Comment by qmonnet** [comment]
Ready to merge, pending TopHat's approval

**Pull request merged**
